tmt/base.py

```diff
@@ -545,20 +546,72 @@ def show(self):
         if self.opt('verbose'):
             self._sources()
 
+    def _lint_execute(self):
+        # execute is required for a plan
+        execute = self.node.get('execute')
+        if not execute:
+            echo(verdict(0, 'execute step must be defined with "how"'))
+            return False
+
+        how = execute.get('how')
+        if how not in ['detach', 'tmt']:
+            echo(verdict(0, f'unsupported execute method "{how}"'))
+            return False
+
+        return True
+
+    def _lint_summary(self):
+        # summary is advised with a resonable length
+        if self.summary is None:
+            echo(verdict(2, 'summary is very useful for quick inspection'))
+        elif len(self.summary) > 50:
+            echo(verdict(2, 'summary should not exceed 50 characters'))
+
+    def _lint_discover(self):
+        # discover step is optional
+        discover = self.node.get('discover')
+        if not discover:
+            return True
+
+        how = discover.get('how')
+        if how not in ['fmf', 'shell']:
+            echo(verdict(0, f'unknown discover method "{how}"'))
+            return False
+
+        return self._lint_discover_fmf(discover)
+
+    @staticmethod
+    def _lint_discover_fmf(discover):
+        # default to `None` for non-existing keys
+        fmf_id = collections.defaultdict(lambda: None, {
+            key: value for key, value in discover.items()
+            if key in ['url', 'ref', 'path'] and value
+            })
+
+        # validate remote id and translate to human readable errors
+        valid, error = tmt.utils.validate_fmf_id(fmf_id)
+
+        if valid:
+            echo(verdict(1, 'fmf remote id is valid'))
+            return True
+
+        echo(verdict(0, error))
+        return False
+
     def lint(self):
         """
         Check plan against the L2 metadata specification.
 
         Return whether the plan is valid.
         """
         self.ls()
-        execute = self.node.get('execute')
-        echo(verdict(execute is not None, 'execute step must be defined'))
-        if self.summary is None:
-            echo(verdict(2, 'summary is very useful for quick inspection'))
-        elif len(self.summary) > 50:
-            echo(verdict(2, 'summary should not exceed 50 characters'))
-        return execute is not None
+
+        # checks not affecting the verdict
+        self._lint_summary()
```

Comment:
What about returning `True` from such checks so that they could be included in `all()` as well?

Reply:
yep, good point. Fixed.